### ISSUE TITLE
Handling drives and folders

### ISSUE THREAD

**Issue by pabloko** [Real-Gecko/Filemin#20]
I've faced some problems to navigate folders built as links or mounted fs, but its easy to fix it by replacing

``` cgi
if ($list[$count - 1][14] eq 'inode/directory' or $list[$count - 1][14] eq 'inode/mount-point' or $list[$count - 1][14] eq 'inode/symlink')
```

on line 196 : filemin-lib.pl

**Comment by Real-Gecko** [comment]
I'll add this to next release, thanks :+1:

**Comment by pabloko** [comment]
Nice, btw this is a great work

Wanted to add table sorting and realized its there but bootstrap.datatable.js rely on jquery.tablesorter.js and must be included on unauthenticated/js folder to work properly.

Its also needed to add this after bootstrap.datatable

``` cgi
print "<link rel=\"stylesheet\" type=\"text/css\" href=\"//cdn.datatables.net/1.10.7/css/jquery.dataTables.min.css\" />";
        print "<script type=\"text/javascript\" src=\"unauthenticated/js/jquery.dataTables.min.js\"></script>";
```

also that quick fix i did need some other enhancement on index.cgi

``` cgi
#filter out folders
    @folders = map {$_} grep {$_->[14] eq 'inode/directory' || $_->[14] eq 'inode/mount-point' || $_->[14] eq 'inode/symlink' } @info;

    #filter out files
    @files = map {$_} grep {$_->[14] ne 'inode/directory' && $_->[14] ne 'inode/mount-point' && $_->[14] ne 'inode/symlink' } @info;

```

in order to list properly the folders in correct sequence.

Also the folder sizes are always 4k wich is useless to have a size column if not display the total size, i did this fix

``` cgi
if ($list[$count - 1][14] eq 'inode/directory' or $list[$count - 1][14] eq 'inode/mount-point' or $list[$count - 1][14] eq 'inode/symlink') {
             $size = `du -hs "$path/$link"`;
             $size =~ s/\/.*//;
```

which will stat the total folder size using "du -dh /path/file" -dh command do the same that nice_size does, also i trim the output to remove the filename that is appended at end.

![thing](http://puu.sh/iJJCq/50140f402d.png)

Great work, keep it going

**Comment by edwardvdv** [comment]
@pabloko I can confirm you fix works for mounted filesystems. Thank you.
@Real-Gecko Thanks for your great work. The current issues with the tired and old Java file manager make sure your piece of code gets a lot of attention :-D

**Comment by Real-Gecko** [comment]
I've used different approach: I'm using Perl's '-d' to determine if entry is file or folder as it was in earlier versions. It works nice with symlinks and mountpoints.
Fixed in version 0.9.3.
